src/connector.js

```diff
@@ -0,0 +1,612 @@
+			const node = document.createElement( "li" );
+			node.setAttribute( "class", "suggestion-item" );
+			node.onclick = ( e ) => { 
+                searchBoxController.selectSuggestion(e.currentTarget.innerText);
+                searchBoxElement.value = e.currentTarget.innerText;
+            };
+			node.innerHTML = suggestion.highlightedValue;
+			suggestionsElement.appendChild( node );
+		});
+		
+		if ( searchBoxState.suggestions.length > 0 ) {
+			suggestionsElement.style.display = 'block';
+		}
+	}
+}
+
+// Filters out dangerous URIs that can create XSS attacks such as `javascript:`.
+function filterProtocol( uri ) {
+
+	const isAbsolute = /^(https?|ftp|file|mailto|tel):/i.test( uri );
+	const isRelative = /^(\/|\.\/|\.\.\/)/.test( uri );
+
+	return isAbsolute || isRelative ? uri : '';
+}
+
+// Clear results list
+function clearResultList() {
+	resultListElement.textContent = "";
+}
+
+// Update results list
+function updateResultListState( newState ) {
+	resultListState = newState;
+	
+	if ( resultListState.isLoading ) {
+		if ( suggestionsElement ) {
+			suggestionsElement.style.display = 'none';
+        }
+	}
+	else {
+		clearResultList();
+		if( !resultListState.hasError && resultListState.hasResults ) {
+			resultListState.results.forEach( ( result, index ) => {
+				const sectionNode = document.createElement( "section" );
+				const highlightedExcerpt = HighlightUtils.highlightString( {
+					content: result.excerpt,
+					highlights: result.excerptHighlights,
+					openingDelimiter: '<strong>',
+					closingDelimiter: '</strong>',
+				} );
+				let itemId = `result-${index}`;
+				const resultDate = new Date( result.raw.date );
+
+				sectionNode.innerHTML = resultTemplateHTML
+					.replace( '%[result.clickUri]', filterProtocol( result.clickUri ) )
+					.replace( 'https://www.canada.ca', filterProtocol( result.clickUri ) ) // hack, invalid href are stripped
+					.replace( '%[result.title]', result.title )
+					.replace( '%[result.raw.author]', result.raw.author ? result.raw.author : '' )
+					.replace( '%[result.breadcrumb]', result.raw.displaynavlabel ? result.raw.displaynavlabel : result.printableUri )
+					.replace( '%[result.printableUri]', result.printableUri)
+					.replace( '%[short-date-en]', resultDate.getFullYear() + "-" + resultDate.getMonth() + 1 + "-" + resultDate.getDate() )
+					.replace( '%[short-date-fr]', resultDate.getFullYear() + "-" + resultDate.getMonth() + 1 + "-" + resultDate.getDate() )
+					.replace( '%[long-date-en]', monthsEn[ resultDate.getMonth() ] + " " + resultDate.getDate() + ", " + resultDate.getFullYear() )
+					.replace( '%[long-date-fr]', resultDate.getDate() + " " + monthsFr[ resultDate.getMonth() ] + " " + resultDate.getFullYear() )
+					.replace( '%[highlightedExcerpt]', highlightedExcerpt );
+				
+				const interactiveResult = buildInteractiveResult( headlessEngine, {
+					options: { result },
+				} );
+
+				let resultLink = sectionNode.querySelector( ".result-link" );
```

Comment:
I suggest to keep that CSS selector configurable in the top of this script.

Reply:
Good point. This is a note for @bblaisATcoveo for his subsequent PR

Reply:
We currently don't have a use case for having this configurable, but note taken for later.